packages/main/src/util.ts

```diff
@@ -18,12 +18,42 @@
 
 import * as fs from 'node:fs';
 import * as os from 'node:os';
+import path from 'node:path';
 
 import { BrowserWindow } from 'electron';
 
-const windows = os.platform() === 'win32';
 export function isWindows(): boolean {
-  return windows;
+  return os.platform() === 'win32';
+}
+export async function isWSL(): Promise<boolean> {
+  const hyperv = await isHyperV();
+  return isWindows() && !hyperv;
+}
+export async function isHyperV(): Promise<boolean> {
+  if (!isWindows()) {
+    return false;
+  }
+  // check if the env variable is set with hyperv
+  if (process.env.CONTAINERS_MACHINE_PROVIDER === 'hyperv') {
+    return true;
+  }
+
+  // as a final step we check if the containers.conf file set the provider to hyperv
+  const providerRegex = /provider\s*=\s*"hyperv"/;
+  return isValueInContainersConfig(providerRegex);
```

Comment:
This is not related to a specific podman machine. It is the general containers config file, and if you set the provider, it will be used as an indication by podman to create a new machine.

If i have to hide/show some properties based on the fact that the user is gonna create a WSL or HyperV instance, i don't have to know which VMtypes are the old machines.

Reply:
BTW i'm gonna move this with the rest of the code to podman extension. Just to clarify

Reply:
for creation of podman machines, toggle like WSL / HyperV or default should be managed by podman extension with the properties on the ConnectionFactory and with enablement/when clause